dev-proxy/ProxyEngine.cs

```diff
@@ -475,10 +475,17 @@ async Task OnRequestAsync(object sender, SessionEventArgs e)
                 await e.GetRequestBodyAsString();
             }
 
-            using var scope = _logger.BeginScope(e.HttpClient.Request.Method ?? "", e.HttpClient.Request.Url, e.GetHashCode());
+            using var scope = _logger?.BeginScope(e.HttpClient.Request.Method ?? "", e.HttpClient.Request.Url, e.GetHashCode());

```

Comment:
I don't think we need to use the null-conditional operator here. We check if _logger isn't null directly in the constructor and it's not-null throughout the code. Let's remove it to simplify the code.